analyzers/src/SonarAnalyzer.Common/Rules/Utilities/UtilityAnalyzerBase.cs

```diff
@@ -76,6 +76,7 @@ protected void ReadParameters(SonarAnalysisContext context, CompilationAnalysisC
     public abstract class UtilityAnalyzerBase<TMessage> : UtilityAnalyzerBase
         where TMessage : IMessage, new()
     {
+        private const int TokenCountThreshold = 7_000;
```

Comment:
I've done a few tests on my PC to see how long it takes to process large number of tokens:

<table>
<tr>
	<td>15000
	<td>~30sec
<tr>
	<td>8500
	<td>~12sec
<tr>
	<td>7000
	<td>~8sec
<tr>
	<td>5500
	<td>~5.5sec
</table>

Reply:
It's not a token count, but identifier count, isn't it?

Does it make sense to highlight tokens if there's more then 1000 of them in a single file from user experience point of view?

Reply:
We count only tokens. The code should be generic enough to cover `SymbolReferenceAnalyzer` too. Which retrieves symbols for all kind of node.

Reply:
> Does it make sense to highlight tokens if there's more then 1000 of them in a single file from user experience point of view?

Probably on large classes it would be more helpful since the code is more complex and hard to read. On the long term I think it makes sense to implement the suggestions from https://github.com/SonarSource/sonar-dotnet/issues/2429#issuecomment-816641431 to be able to increase this limit.